lib/matplotlib/image.py

```diff
@@ -168,11 +169,37 @@ def _resample(
     allocating the output array and fetching the relevant properties from the
     Image object *image_obj*.
     """
+
+    # decide if we need to apply anti-aliasing if the data is upsampled:
+    # compare the number of displayed pixels to the number of
+    # the data pixels.
+    interpolation = image_obj.get_interpolation()
+    if interpolation == 'antialiased':
+        # don't antialias if upsampling by an integer number or
+        # if zooming in more than a factor of 3
+        # do antialiasing....
+        # compare the number of displayed pixels of the image to the number of
+        # the data pixels.
```

Comment:
perhaps this comment block can be slightly shortened/combined with the block above